src/modules/0.0.22/aws_codedeploy/mappers/deployment.ts

```diff
@@ -18,15 +18,29 @@ import { CodedeployDeployment, DeploymentStatusEnum, RevisionType } from '../ent
 export class CodedeployDeploymentMapper extends MapperBase<CodedeployDeployment> {
   module: AwsCodedeployModule;
   entity = CodedeployDeployment;
+
   equals = (a: CodedeployDeployment, b: CodedeployDeployment) => {
+    let locationEqual: boolean;
+    if (a.location?.revisionType === b.location?.revisionType) {
+      if (a.location?.revisionType === RevisionType.GITHUB) {
+        locationEqual =
+          Object.is(a.location.githubLocation?.repository, b.location?.githubLocation?.repository) &&
+          Object.is(a.location.githubLocation?.commitId, b.location?.githubLocation?.commitId);
+      } else {
+        locationEqual =
+          Object.is(a.location?.s3Location?.bucket, b.location?.s3Location?.bucket) &&
+          Object.is(a.location?.s3Location?.key, b.location?.s3Location?.key);
+      }
+    } else return false;
+
     return (
       isEqual(a.application.name, b.application.name) &&
       isEqual(a.deploymentGroup.name, b.deploymentGroup.name) &&
       Object.is(a.deploymentId, b.deploymentId) &&
       Object.is(a.description, b.description) &&
       Object.is(a.externalId, b.externalId) &&
       Object.is(a.status, b.status) &&
-      isEqual(a.location, b.location)
+      locationEqual
```

Comment:
Nit: `Object.is` correctly marks `Object.is(undefined, undefined)` as `true`

![Screenshot from 2022-10-17 14-04-29](https://user-images.githubusercontent.com/765551/196261122-fd4ea927-eb24-4881-a64f-5bb4df29b02e.png)

So all of this can be switched to simply:

```ts
return (
  isEqual(a.application.name, b.application.name) &&
  isEqual(a.deploymentGroup.name, b.deploymentGroup.name) &&
  Object.is(a.deploymentId, b.deploymentId) &&
  Object.is(a.description, b.description) &&
  Object.is(a.externalId, b.externalId) &&
  Object.is(a.status, b.status) &&
  Object.is(a.location?.revisionType, b.location?.revisionType) &&
  Object.is(a.location?.githubLocation?.repository, b.location?.githubLocation?.repository) &&
  Object.is(a.location?.githubLocation?.commitId, b.location?.githubLocation?.commitId) &&
  Object.is(a.location?.s3Location?.bucket, b.location?.s3Location?.bucket) &&
  Object.is(a.location?.s3Location?.key, b.location?.s3Location?.key)
);
```